Sources/NIOKit/Extensions/Transform.swift

```diff
@@ -0,0 +1,152 @@
+import NIO
+
+extension EventLoopFuture {
+    /// Maps the current future to contain the new type. Errors are carried over, successful (expected) results are transformed into the given instance.
+    ///
+    ///     user.save(on: req).transform(to: HTTPStatus.created)
+    ///
+    public func transform<T>(to instance: T) -> EventLoopFuture<T> {
+        return self.map { _ in
+            instance
+        }
+    }
+    
+    /// Maps the current future to contain the new type. Errors are carried over, successful (expected) results are transformed into the given instance.
+    ///
+    ///     let user = User.find(id, on: request)
+    ///     posts.save(on: request).transform(to: user)
+    ///
+    public func transform<T>(to future: EventLoopFuture<T>) -> EventLoopFuture<T> {
+        return self.then { _ in
+            future
+        }
+    }
+}
+
+/// Maps the current futures to contain the new type. Errors are carried over, successful (expected) results are transformed into the given instance.
+///
+///     transform(futureA, futureB, to: HTTPStatus.created)
+///
+public func transform<A, B, Result>(
```

Comment:
As in, remove the function alltogether and instead of calling `transform(futureaA, butureB, to: .created)` just have users call `futureA.and(futureB).transform(to: .created)` or are you talking about the inside of the function? :)

Reply:
Bump :)

Reply:
I think remove. In general, I think we should move toward using `EventLoopFuture.and` for anything we needed multiple pseudo-variadic generic methods for before.

Reply:
I would agree. I'll fix it :)